lib/git/base.rb

```diff
@@ -313,8 +313,8 @@ def checkout_file(version, file)
 
     # fetches changes from a remote branch - this does not modify the working directory,
     # it just gets the changes from the remote if there are any
-    def fetch(remote = 'origin', opts={})
-      self.lib.fetch(remote, opts)
+    def fetch(remote = 'origin', ref = nil, opts={})
```

Comment:
This could potentially break implementations currently using `#fetch`. For example,

```ruby
g.fetch(origin, option: "something")
```

Will set `remote` to `origin` and `ref` to `{ option: "something"}`

Maybe we can utilize it inside of `opts`?

Reply:
Good point, will look into this.